src/flows/pipes/Region/view.tsx

```diff
@@ -0,0 +1,301 @@
+import React, {FC, useContext, useEffect, useState, useMemo} from 'react'
+import {useSelector} from 'react-redux'
+
+import {activeFlags} from 'src/shared/selectors/flags'
+import {PipeProp} from 'src/types/flows'
+import {PipeContext} from 'src/flows/context/pipe'
+
+import {
+  Dropdown,
+  ComponentColor,
+  ComponentSize,
+  Form,
+  FlexBox,
+  Input,
+  InputType,
+  ComponentStatus,
+} from '@influxdata/clockface'
+
+import './style.scss'
+
+const REGIONS = [
+  {
+    label: 'US West (Oregon)',
+    group: 'Amazon Web Services',
+    value: 'https://us-west-2-1.aws.cloud2.influxdata.com',
+  },
+  {
+    label: 'US East (Virginia)',
+    group: 'Amazon Web Services',
+    value: 'https://us-east-1-1.aws.cloud2.influxdata.com',
+  },
+  {
+    label: 'Europe West (Frankfurt)',
+    group: 'Amazon Web Services',
+    value: 'https://eu-central-1-1.aws.cloud2.influxdata.com',
+  },
+  {
+    label: 'US Central (Iowa)',
+    group: 'Google Cloud Platform',
+    value: 'https://us-central1-1.gcp.cloud2.influxdata.com',
+  },
+  {
+    label: 'Europe West (Belgium)',
+    group: 'Google Cloud Platform',
+    value: 'https://europe-west1-1.gcp.cloud2.influxdata.com',
+  },
+  {
+    label: 'US East (Virginia)',
+    group: 'Microsoft Azure',
+    value: 'https://eastus-1.azure.cloud2.influxdata.com',
+  },
+  {
+    label: 'Europe West (Amsterdam)',
+    group: 'Microsoft Azure',
+    value: 'https://westeurope-1.azure.cloud2.influxdata.com',
+  },
+  {
+    label: 'Tools Cluster',
+    group: 'Development',
+    flag: 'local-dev',
+    value: 'https://influxdb.aws.influxdata.io',
+  },
+  {
+    label: 'Local Kind Cluster',
+    group: 'Development',
+    flag: 'local-dev',
+    value: 'https://twodotoh.a.influxcloud.dev.local',
+  },
+  {label: 'Self Hosted', value: 'self-hosted'},
+]
+
+const Source: FC<PipeProp> = ({Context}) => {
+  const {data, update} = useContext(PipeContext)
+  const [error, setError] = useState<boolean>(false)
+  const flags = useSelector(activeFlags)
+
+  const updater = option => {
+    if (option.value === 'self-hosted') {
+      update({
+        source: 'custom',
+        region: 'https://localhost:8086',
```

Comment:
should this be based on anything in the dev env?

Reply:
nope. it can be cloud hosted and is run relative to the browser. its up to the user to ensure that this url is available to their task they generate